### PR TITLE
first pass review of projects page

### PR DESCRIPTION
**Pull request by ykl7**
Reviewing content on group projects. Still a work in progress.

### CODE REVIEW

**Comment by leafac** [comment]
Please pull often, I’m touching these files a lot.

**Comment by ykl7** [comment]
@leafac I don't see any mention of testing in any of the pages. Everything focuses on the features. I think we should make students focus on testing in  Iteration 4, along with writing some more features. Right now, looking at the pages gives the impression that they need to have a hundred features in their application (after all, there are 3 iterations for it). Plus, the final iteration implicitly entails finishing up the non-CRUD feature anyways.

File: iterations/0.md [hunk 1]
```diff
@@ -159,11 +159,11 @@ If it’s clear that the project is viable (for example, if you’re using only
 
 Determine whether the project you’re proposing is at the right scale.
 
-This is subjective and difficult to estimate, so you must talk to the staff about it. At the bare minimum your project must include one _non-data-manipulation_ feature; ideally it includes several.
+This is subjective and difficult to estimate, so you must talk to the staff about it. At the bare minimum your project must include one _non-data-manipulation (non-CRUD)_ feature; ideally it includes several.
```
Comment (leafac) [inline iterations/0.md]: CRUD is jargon. Either we introduce and explain it, or we don’t use it. I’d prefer not to use it.

File: iterations/2.md [hunk 2]
```diff
@@ -15,13 +15,13 @@ Discuss the design of your application with your advisor. You may need to revise
 # First Feature
 
 <small>
-You may need to revise parts of the [Design](/iterations/1#design), and even parts of the [Project Proposal](/iterations/0#project-proposal) as you develop the application.
+You may need to revise parts of the [Design](/iterations/1#design), and even parts of the [Project Proposal](/iterations/0#project-proposal) as you develop the application. Remember to update the documents as required.
```
Comment (leafac) [inline iterations/2.md]: I like this, but please cut the “remember to.” Just say “Update the documents as required.” Also, please propagate this change to the other iterations (which repeat this sentence).

File: iterations/2.md [hunk 3]
```diff
@@ -15,13 +15,13 @@ Discuss the design of your application with your advisor. You may need to revise
...
 </small>
 
-You must implement at least the first feature of your application from end to end. Typically this is one of the central features of your application, but it must also be something relatively simple, because it’s your first feature. The feature must exercise every part of your [Architecture](/iterations/1#architecture), including database, tests, API design, deployment, documentation, and so forth. You must work with your advisor early in the iteration to establish what’s a suitable first feature.
+You must implement at least the first feature of your application from end to end. Typically, this is one of the central features of your application, but it should be something relatively simple, because it’s your first feature. The feature must exercise every part of your [Architecture](/iterations/1#architecture), including database, tests, API design, deployment, documentation, and so forth. It is a good idea to work with your advisor **early** in the iteration to establish what’s a suitable first feature.
```
Comment (leafac) [inline iterations/2.md]: I don’t like the changes. Going from “you must” to “it is a good idea” waters the message down, which is why you had to introduce the bold. I’d just revert this line to the original.

By the way, [we’re using contractions](https://www.plainlanguage.gov/guidelines/conversational/use-contractions/), so it should have been “it’s a good idea”, not “it is a good idea.” Also, note the [curly quote](https://practicaltypography.com/straight-and-curly-quotes.html) I put there when writing the contraction.

File: iterations/3.md [hunk 4]
```diff
@@ -12,6 +12,6 @@ You may need to revise parts of the [Design](/iterations/1#design), and even par
 Submit the code base for of your application in the root directory of the `master` branch of your group’s repository at `https://github.com/jhu-oose/{{site.course}}-group-<identifier>`.
 </small>
 
-You must implement the core _distinguishing_ features of your application and demonstrate them on the [Midterm Presentation](/group-projects#presentations) by the end of the iteration. The application doesn’t have to be usable yet, but it must be working well enough to show the core concepts in action. What this means depends a lot on the nature of your project, so you must work with your advisor early in the iteration to establish the goals.
+You must implement the core _distinguishing_ features of your application and demonstrate them on the [Midterm Presentation](/group-projects#presentations) by the end of the iteration. The application doesn’t have to be usable yet, but it must be working well enough to show the core concepts in action. The specifics of what this entails depends a lot on the nature of your project, so you must work with your advisor early in the iteration to establish the goals.
```
Comment (leafac) [inline iterations/3.md]: Lovely, please propagate to the next iterations pages, which contain a similar message.

File: iterations/0.md (hunk 1 above)
Comment (ykl7) [reply iterations/0.md]: Changed

File: iterations/3.md (hunk 4 above)
Comment (ykl7) [reply iterations/3.md]: Done

File: iterations/2.md (hunk 3 above)
Comment (ykl7) [reply iterations/2.md]: Alright, I'll keep the **early** and let go of all other changes.

File: iterations/2.md (hunk 2 above)
Comment (ykl7) [reply iterations/2.md]: Resolved

File: iterations/0.md (hunk 1 above)
Comment (leafac) [reply iterations/0.md]: Did you push? I still see the change…

Comment (ykl7) [reply iterations/0.md]: I just pushed with the changes we talked about in these comments. I'm still writing up the importance of testing for each feature.

File: iterations/2.md (hunk 2 above)
Comment (leafac) [reply iterations/2.md]: Please propagate this change to the other iterations (which repeat this sentence).

**Comment by ykl7** [comment]
Have a look now @leafac

**Comment by leafac** [comment]
I changed a small thing here and there and merged it. Thank you very much!

**Pull request merged**